tests/internals/trigger_update_so/trigger_update_so_test.go

```diff
@@ -0,0 +1,426 @@
+		"replica count should be %d after 3 minutes", minReplicas)
+
+	testTargetValue(t, kc, data)          // one trigger target changes
+	testTwoTriggers(t, kc, data)          // add trigger during active scaling
+	testRemoveTrigger(t, kc, data)        // remove trigger during active scaling
+	testThreeTriggersWithCPU(t, kc, data) // three triggers
+
+	DeleteKubernetesResources(t, kc, namespace, data, templates)
+}
+
+// tests basic scaling with one trigger based on metrics
+func testTargetValue(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- test target value 1 ---")
+	KubectlApplyWithTemplate(t, data, "scaledObjectTriggerTemplate", scaledObjectTriggerTemplate)
+
+	data.MetricValue = 1
+	KubectlApplyWithTemplate(t, data, "updateMetricTemplate", updateMetricTemplate)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, 1, 180, 3),
+		"replica count should be %d after 3 minutes", 1)
+
+	t.Log("--- test target value 10 ---")
+	data.MetricValue = 10
+	KubectlApplyWithTemplate(t, data, "updateMetricTemplate", updateMetricTemplate)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, maxReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", maxReplicas)
+
+	t.Log("--- test target value 0 ---")
+	data.MetricValue = 0
+	KubectlApplyWithTemplate(t, data, "updateMetricTemplate", updateMetricTemplate)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, minReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", minReplicas)
+}
+
+// test adding second trigger during scaling
+func testTwoTriggers(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- test two triggers ---")
+	KubectlApplyWithTemplate(t, data, "scaledObjectTriggerTemplate", scaledObjectTriggerTemplate)
+
+	data.MetricValue = 1
+	KubectlApplyWithTemplate(t, data, "updateMetricTemplate", updateMetricTemplate)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, 1, 180, 3),
+		"replica count should be %d after 3 minutes", 1)
+
+	KubectlApplyWithTemplate(t, data, "scaledObjectTwoTriggerTemplate", scaledObjectTwoTriggerTemplate)
+	// scale to max with k8s wl = second trigger
+	KubernetesScaleDeployment(t, kc, workloadDeploymentName, int64(maxReplicas), namespace)
+
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, namespace, maxReplicas, 180, 3),
+		"replica count should be %d after 3 minutes", maxReplicas)
+}
+
+// scales to max with kubernetes worload(second trigger), removes it and
+// scales to 3 replicas based on metric value (first trigger)
+func testRemoveTrigger(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- test remove trigger 2 -> 1 ---")
+	KubectlApplyWithTemplate(t, data, "scaledObjectTwoTriggerTemplate", scaledObjectTwoTriggerTemplate)
+	data.MetricValue = 5 // 3 replicas (midReplicas)
```

Comment:
```suggestion
	data.MetricValue = 5 // 3 replicas (minReplicas)
```

Reply:
this is intentional, setting MetricValue to 5 scales to 3 replicas which is not minReplicas (Im scaling up and then testing the scenario)